algo-core/scheduler_utils.py

```diff
@@ -0,0 +1,156 @@
+import argparse
+import json
+import sys
+from pathlib import Path
+import jsonschema
+import pandas as pd
+
+
+# Input filenames:
+filename_onboarding = "onboarding_agents.txt"
+filename_mentors = "mentors.txt"
+filename_new = "new_agents.txt"
+input_folder = ""
+
+def get_project_root() -> str:
+    return str(Path(__file__).parent.parent)
+
+
+def parse_json_input():
+    """Read, validate and return json input."""
+    # Production (read input from command line):
+    global input_folder
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-i", "--input", help="Scheduler input JSON file path", required=True
+    )
+    args = parser.parse_args()
+    input_filename = args.input.strip()
+
+    # Testing (define input directly):
+    # input_filename = 'support-shift-scheduler-input.json'
+
+    # Load and validate JSON input:
+    input_json = json.load(open(input_filename))
+    input_json_schema = json.load(
+        open(get_project_root() + "/lib/schemas/support-shift-scheduler-input.schema.json")
+    )
+    try:
+        jsonschema.validate(input_json, input_json_schema)
+    except jsonschema.exceptions.ValidationError as err:
+        print("Input JSON validation error", err)
+        sys.exit(1)
+
+    input_folder = get_project_root() + '/logs/' + input_json["options"]["startMondayDate"] + '_' + input_json["options"]["modelName"] + '/'
```

Comment:
This is another good example where the `Path` object would shine IMO

```suggestion
    input_folder = Path(get_project_root(), 'logs', f'{input_json["options"]["startMondayDate"]}_{input_json["options"]["modelName"]}')
```
or, assuming that `get_project_root()` returns a `Path` object,
```suggestion
    input_folder = get_project_root() / 'logs' / f'{input_json["options"]["startMondayDate"]}_{input_json["options"]["modelName"]}'
```

both are equivalent and takes care of the separators automatically, making the code cross-platform.

Reply:
maybe that was the issue I had before, I'll check it out!